contracts/teller-network/Escrow.sol

```diff
@@ -69,6 +79,63 @@ contract Escrow is Pausable, MessageSigned, Fees, Arbitrable {
     event Rating(uint indexed offerId, address indexed buyer, uint indexed escrowId, uint rating, uint date);
 
 
+    mapping(address => uint) public lastActivity;
+
+    function accept_relayed_call(address relay, address from, bytes memory encoded_function, uint gas_price,
+      uint transaction_fee ) public view returns(uint32) {
+
+        bytes4 fSign = abiDecodeFunctionSignature(encoded_function);
+
+        if(fSign != CREATE_SIGNATURE && fSign != PAY_SIGNATURE && fSign != CANCEL_SIGNATURE && fSign != OPEN_CASE_SIGNATURE)
+            return 11;
+            
+        if(from.balance > 500000 * gas_price) return 12; // According to tests, 333450 is the cost of creating an escrow, so 500000 should be good
+
+        // Only allow trxs where the user is a buyer
+        if(fSign == PAY_SIGNATURE || fSign == CANCEL_SIGNATURE || fSign == OPEN_CASE_SIGNATURE){
+            bytes memory escrowBytes = slice(encoded_function, 4, 32);
+            uint escrowId;
+            assembly {
+                escrowId := mload(add(escrowBytes, add(0x20, 0)))
+            }
+            if(escrowId >= transactions.length) return 13;
+            if(transactions[escrowId].buyer != from) return 14;
+            if(metadataStore.getAsset(transactions[escrowId].offerId) != address(0)) return 15; // Must be eth trx
+
+            if(fSign == CANCEL_SIGNATURE){ // Allow activity after 15min have passed
+                if((lastActivity[from] + 15 minutes) > block.timestamp) return 17;
```

Comment:
Why do we do that? So we block spammers?

Reply:
yeah... it's a weak way to slow down spammers.

Reply:
Is there a way it could block real users too? Will we show a good message for it or will they just have a generic error?

Reply:
It will block real users too, so a good error message is required